Dynatrace/dynatrace-operator
# Skip OneAgent injection in case status is not ready

## Description


To prevent pods from getting stuck the webhook does no longer inject into user applications in the scenario when the state is unknown.

[Jira](https://dt-rnd.atlassian.net/browse/K8S-11367)

## How can this be tested?

- Apply CNFS DynaKube 
- Delete the `tenantUUID` from the status
  - Did not manage to delete it using `kubectl`. Instead I used `kubectl proxy` and `curl -XPATCH -H "Content-Type: application/merge-patch+json" \
--data '{"status": {"oneAgent": {"connectionInfoStatus": {"tenantUUID": null}}, "activeGate": {"connectionInfoStatus": {"tenantUUID": null}}}}' \
http://localhost:8001/apis/dynatrace.com/v1beta3/namespaces/dynatrace/dynakubes/dynakube/status`
- Create some pods that should be monitored
- Check if they were OneAgent injected and their annotations
